### DISCUSSION TITLE
Error processing file

### DISCUSSION THREAD

**Discussion by LizFun** [danny-avila/LibreChat#2054] [Troubleshooting]
### What happened?

As the title suggests, an error occurred when I was uploading a PDF file, halfway through the upload process.

Error log：
      error: [/files] Error processing file: handleFileUpload is not a function

### Steps to Reproduce

1. Select openAI
2. Uploading a PDF file
3. Error occurred

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
OpenAI does not yet support PDF upload. Only the Assistants endpoint after one is selected.

**Reply by aodrasa** [reply]
Assuming the API supports different files than the ChatGPT app correct? What formats are accepted by the API? I'm having trouble with .doc, .csv and .xlxs

**Reply by danny-avila** [reply]
All of those should work. Also there is now general file support for almost every endpoint including custom ones now, following this PR: https://github.com/danny-avila/LibreChat/pull/2143

It's in beta state as there is still a few things to iron out, see the PR for details on how to setup

**Comment by mikaelaldy** [comment]
same but I use openrouter API with sonnet-3.5. 
![Screenshot (6)](https://github.com/user-attachments/assets/9a937499-7355-459e-aa30-53e4e245113f)

**Reply by odin-loki** [reply]
I'm having problems too. I mean I just want to dev.

**Comment by Medenor** [comment]
I encounter the same issue with OpenAI, Perplexity, Gemini, Claude, and various models accessed through OpenRouter, regardless of the file type.
